shell/core/helpers.js

```diff
@@ -0,0 +1,139 @@
+import { ActionLocation, CardLocation, ExtensionPoint } from '@shell/core/types';
+import { isMac } from '@shell/utils/platform';
+import { ucFirst } from '@shell/utils/string';
+
+function checkExtensionRouteBinding({ name, params, query }, locationConfig) {
+  // console.log('name && params', name, params);
+
+  // if no configuration is passed, consider it as global
+  if (!Object.keys(locationConfig).length) {
+    return true;
+  }
+
+  // "params" to be checked based on the locationConfig
+  const paramsToCheck = [
+    'product',
+    'resource',
+    'namespace',
+    'cluster',
+    'id',
+    'mode'
+  ];
+
+  let res = false;
+
+  for (let i = 0; i < paramsToCheck.length; i++) {
+    const param = paramsToCheck[i];
+
+    if (locationConfig[param]) {
+      // handle "product" in a separate way...
+      if (param === 'product') {
+        if (locationConfig[param] === 'home' && !name.startsWith('c-')) {
+          res = true;
+        } else if (locationConfig[param] === 'explorer' && name.startsWith('c-')) {
+          res = true;
+        } else if (locationConfig[param] === params[param]) {
+          res = true;
+        } else {
+          res = false;
+          break;
+        }
+        // also handle "mode" in a separate way because it mainly depends on query params
+      } else if (param === 'mode') {
+        if (locationConfig[param] === 'edit' && query.mode && query.mode === 'edit') {
```

Comment:
There's some `const` that can replace these from `query-params.js`. `_EDIT`, `_CONFIG`, etc

Reply:
Fixed